Zend/zend_ini_parser.y

```diff
@@ -440,7 +444,9 @@ expr:
 ;
 
 cfg_var_ref:
-		TC_DOLLAR_CURLY TC_VARNAME '}'	{ zend_ini_get_var(&$$, &$2); zend_string_free(Z_STR($2)); }
+		TC_DOLLAR_CURLY TC_VARNAME TC_COLON_DASH TC_RAW '}'	{ zend_ini_get_var(&$$, &$2, &$4); zend_string_free(Z_STR($2)); zend_string_free(Z_STR($4)); }
+	|	TC_DOLLAR_CURLY TC_VARNAME TC_COLON_DASH '}'		{ zend_ini_get_var(&$$, &$2, NULL); zend_string_free(Z_STR($2)); }
```

Comment:
I couldn't get the fallback value to be optional (I want `${foo:-}` to work the same as `${foo}`) without writing two rules. I wonder if there's a way to simplify this.

Reply:
You can try add two new productions.

```yacc
cfg_var_ref:
		TC_DOLLAR_CURLY TC_VARNAME fallback '}'	{
			zend_ini_get_var(&$$, &$2, $3);
			zend_string_free(Z_STR($2));
			if ($3) { zend_string_free(Z_STR($3)); }
		}
;

fallback:
		TC_COLON_DASH fallback_value { $$ = $2; }
	|	%empty { $$ = NULL; }
;

fallback_value:
		TC_RAW { $$ = $1; }
	|	%empty { $$ = NULL; }
;
```

Reply:
Hmm I can't seem to make it accept `{ $$ = NULL }`. I tried casting to `zval *`, but `$$` expects a `zval`. I searched around, but can't seem to find how to easily make a null `zval`. Can you point me in the right direction?

```
php-src/Zend/zend_ini_parser.y:455:32: error: assigning to 'INI_STYPE' (aka 'struct _zval_struct') from incompatible type 'void *'
                       { yyval = NULL; }
                               ^ ~~~~
php-src/Zend/zend_ini_parser.y:459:32: error: assigning to 'INI_STYPE' (aka 'struct _zval_struct') from incompatible type 'void *'
                       { yyval = NULL; }
                               ^ ~~~~
```

Reply:
I forgot we're dealing with `zval` instead of `zval*` here. You could try `ZVAL_UNDEF(&$$)` instead of `$$ = NULL;` and `Z_TYPE($$) == IS_UNDEF` instead of `$$ == NULL;`.

Reply:
That worked, thanks!